### DISCUSSION TITLE
whats the correct way of extending cue with custom builtins ?

### DISCUSSION THREAD

**Discussion by ymolists** [cuelang/cue#1052] [Q&A]
I am trying to see if i can build extensions for cue that dont need to live in the cue main repo. Something as a makefile replacement for example. All the make rules would be defined as cue structures. Its a toy example of course.

I am imagining that i would need to add a new set of builtins added. How would i go about ? anyone done this before ?

**Comment by myitcv** [comment]
Please see discussion in https://github.com/cuelang/cue/issues/350

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/1052.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
